native/src/tablecloth.mli

```diff
@@ -231,84 +231,234 @@ end
 
 module Tuple2 : sig
   val create : 'a -> 'b -> 'a * 'b
+  (** Create a 2-tuple.
+
+    Tuple2.create 3 4 == (3, 4)
+
+    let zip (xs : 'a list) (ys : 'b list) : ('a * 'b) list =
+      List.map2 Tuple2.create xs ys
+  *)
 
   val first : ('a * 'b) -> 'a
+  (** Extract the first value from a tuple.
+
+    Tuple2.first (3, 4) == 3
+    Tuple2.first ("john", "doe") == "john"
+  *)
 
   val second : ('a * 'b) -> 'b
+  (** Extract the second value from a tuple.
+
+    Tuple2.second (3, 4) == 4
+    Tuple2.second ("john", "doe") == "doe"
+  *)
 
   val mapFirst : f:('a -> 'x) -> ('a * 'b) -> ('x * 'b)
+  (** Transform the first value in a tuple.
+
+    Tuple2.mapFirst ~f:String.reverse ("stressed", 16) == ("desserts", 16)
+    Tuple2.mapFirst ~f:String.length ("stressed", 16) == (8, 16)
+  *)
 
   val map_first : f:('a -> 'x) -> ('a * 'b) -> ('x * 'b)
 
   val mapSecond : f:('b -> 'y) -> ('a * 'b) -> ('a * 'y)
+  (** Transform the second value in a tuple.
+
+    Tuple2.mapSecond ~f:sqrt ("stressed", 16.) == ("stressed", 4.)
+    Tuple2.mapSecond ~f:(~-) ("stressed", 16) == ("stressed", -16)
```

Comment:
Is this syntax valid (`~-`)? I would have though it would be `(-)`

Reply:
Yep, it's the 'negate' operator for integers: https://caml.inria.fr/pub/docs/manual-ocaml/libref/Pervasives.html#1_Integerarithmetic.

We should add an Int.negate to Tablecloth!

Reply:
Ah, cool. Yes we should!